components/UI/FilterMenu/FilterTimeMenu.tsx

```diff
@@ -91,6 +97,19 @@ const FilterTimeMenu: React.FC<FilterTimeMenuProps> = ({ onChangeTime }) => {
     handleClose();
   };
 
+  useEffect(() => {
+    let currentOptionIndex = FilterOptions.findIndex(
+      (option) => option.value === selectedValue
+    );
+
+    if (shouldFetchNextOption && selectedValue !== "all") {
+      setSelectedValue(FilterOptions[currentOptionIndex + 1].value);
+      resetShouldFetchNextOption();
+    }
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Why do we need to disable that rule here?

Reply:
It breaks functionality when ESLint insists on adding other variables to useEffect that shouldn't be there.